SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -2151,212 +2151,146 @@ void CNEMOEulerSolver::BC_Outlet(CGeometry *geometry, CSolver **solver_container
-//                                           geometry->nodes->GetCoord(iPoint), Coord_Reflected);
-//        visc_numerics->SetCoord(geometry->nodes->GetCoord(iPoint), Coord_Reflected);
-//
-//        /*--- Primitive variables, and gradient ---*/
-//        visc_numerics->SetPrimitive(V_domain, V_inlet);
-//        visc_numerics->SetPrimVarGradient(nodes->GetGradient_Primitive(iPoint), nodes->GetGradient_Primitive(iPoint));
-//
-//        /*--- Laminar viscosity ---*/
-//        visc_numerics->SetLaminarViscosity(nodes->GetLaminarViscosity(iPoint), nodes->GetLaminarViscosity(iPoint));
-//
-//        /*--- Compute and update residual ---*/
-//        auto residual = visc_numerics->ComputeResidual(config);
-//        LinSysRes.SubtractBlock(iPoint, residual);
-//
-//        /*--- Jacobian contribution for implicit integration ---*/
-//        //if (implicit)
-//        //  Jacobian.SubtractBlock2Diag(iPoint, residual.Jacobian_i);
-//      }
-//
-//    }
-//  }
-//
-//  /*--- Free locally allocated memory ---*/
-//  delete [] U_domain;
-//  delete [] U_inlet;
-//  delete [] V_domain;
-//  delete [] V_inlet;
-//  delete [] Normal;
+ unsigned short iDim;
+ unsigned long iVertex, iPoint, Point_Normal;
+
+ bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+ bool dynamic_grid  = config->GetGrid_Movement();
+ bool viscous       = config->GetViscous();
+ string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+
+ su2double *Normal = new su2double[nDim];
+
+ /*--- Supersonic inlet flow: there are no outgoing characteristics,
+  so all flow variables can be imposed at the inlet.
+  First, retrieve the specified values for the primitive variables. ---*/
+
+ const su2double* Mass_Frac       = config->GetInlet_MassFrac(); 
+ const su2double Temperature      = config->GetInlet_Temperature(Marker_Tag);
+ const su2double Pressure         = config->GetInlet_Pressure(Marker_Tag);
+ const su2double* Velocity        = config->GetInlet_Velocity(Marker_Tag);
+ su2double Temperature_ve   = config->GetInlet_Temperature_ve();
+
+ /*--- If no Tve value specified (left as 0 K default), set to Ttr value ---*/
+ if (Temperature_ve == 0.0) {Temperature_ve = Temperature;}
+
+ /*--- Set mixture state ---*/
+ FluidModel->SetTDStatePTTv(Pressure, Mass_Frac, Temperature, Temperature_ve);
+
+ /*--- Compute Ma vector for flow direction ---*/
+ const su2double soundspeed = FluidModel->ComputeSoundSpeed();
+
+ su2double* Mvec = new su2double[nDim];
+
+ for (iDim = 0; iDim < nDim; iDim++)
+   Mvec[iDim] = Velocity[iDim] / soundspeed;
+
+ /*--- Allocate inlet node to compute gradients for numerics ---*/
+ node_inlet = new CNEMOEulerVariable(Pressure, Mass_Frac, Mvec,
+                                        Temperature, Temperature_ve,
+                                        1, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                        config, FluidModel);
```

Comment:
This is not very good because it will allocate a bunch of stuff everytime, either persist node_inlet, only allocating it once, or make it a local variables as shown below.
```suggestion
 CNEMOEulerVariable node_inlet(Pressure, Mass_Frac, Mvec,
                                        Temperature, Temperature_ve,
                                        1, nDim, nVar, nPrimVar, nPrimVarGrad,
                                        config, FluidModel);
```